app/controllers/links_controller.rb

```diff
@@ -32,53 +33,57 @@ def create
 
     respond_to do |format|
       if @link.save
-        format.html { redirect_to root_path, notice: 'Link was successfully created.' }
-        format.js 
+        format.html do
+          redirect_to root_path, notice: "Link was successfully created."
+        end
       else
-        format.html { render :form, notice: "Please paste the full URL (with http)" }
-        format.js
-        # format.json { render json: @link.errors, status: :unprocessable_entity }
+        format.html do
+          redirect_to root_path, alert: "Please enter a valid URL (with http)"
+        end
       end
+      format.js
     end
   end
 
   def update
     respond_to do |format|
       if @link.update(link_params)
-        format.html { redirect_to :dashboard, notice: 'Link updated successfully.' }
+        format.html do
+          redirect_to :dashboard, notice: "Link updated successfully."
+        end
       else
-        format.html { redirect_to :dashboard, notice: 'Error occured' }
+        format.html { redirect_to :dashboard, notice: "Error occured" }
         format.js
-        format.json { render json: @link.errors, status: :unprocessable_entity }
       end
     end
   end
 
-  def manage
-    @link = Link.new
-    @links = Link.where(user_id: current_user.id).order(created_at: :desc)
-    render "index"
-  end
-    
   def destroy
     @link.destroy
     respond_to do |format|
-      format.html { redirect_to :dashboard, notice: 'Link deleted successfully.' }
-      format.json { head :no_content }
+      format.html do
+        redirect_to :dashboard, notice: "Link deleted successfully."
+      end
     end
   end
 
   private
-    def set_link
-      @link = Link.find(params[:id])
-    end
 
-    # Never trust parameters from the scary internet, only allow the white list through.
-    def new_link_params
-      params.require(:link).permit(:original)
-    end
+  def set_link
+    @link = Link.find(params[:id])
+  end
 
-    def link_params
-      params.require(:link).permit(:original, :slug, :active)
+  def my_links
+    if current_user
+      @links = Link.where(user_id: current_user.id).order(created_at: :desc)
     end
+  end
+
+  def new_link_params
+    params.require(:link).permit(:original, :slug)
+  end
+
+  def link_params
+    params.require(:link).permit(:original, :slug, :active)
```

Comment:
**Critical - Code Functionality**
Why do you have to define both `link_params` and `new_link_params`?
You could just make use of `link_prams`